endpoints.go

```diff
@@ -33,3 +42,402 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
+        return
+    }
+
+    var survey models.Survey
+
+    err = json.Unmarshal(body, &survey)
+    if err != nil {
+        http.Error(w, "Error unmarshalling JSON", http.StatusBadRequest)
+        return
+    }
+
+    if survey.ShortName == "" && survey.LongName == "" && survey.LegalBasis == "" && survey.SurveyMode == "" {
+        http.Error(w, "No values to update", http.StatusBadRequest)
+        return
+    }
+
+    tx, err := db.Begin()
+    if err != nil {
+        http.Error(w, "Error starting database transaction", http.StatusInternalServerError)
+        return
+    }
+    defer tx.Rollback()
+
+    results := db.QueryRow("SELECT * FROM " + viper.GetString("db_schema") + ".survey WHERE survey_ref = $1" , params["surveyRef"])
+    var tempSurvey models.Survey
+    err = results.Scan(&tempSurvey.ID, &tempSurvey.SurveyRef, &tempSurvey.ShortName, &tempSurvey.LongName, &tempSurvey.LegalBasis, &tempSurvey.SurveyMode)
+    if err != nil {
+        if err == sql.ErrNoRows {
+            http.Error(w, "Survey reference not found", http.StatusNotFound)
+            return
+        }
+        http.Error(w, "Check query failed", http.StatusInternalServerError)
+        return
+    }
+
+    newShortName := "short_name"
+    newLongName := "long_name"
+    newLegalBasis := "legal_basis"
+    newSurveyMode := "survey_mode"
+
+    //If the JSON body filled in a field, update the value in the database, else put the un-updated value into the survey object so that it's part of the returned JSON
+    if survey.ShortName != "" {
+        newShortName = survey.ShortName
+    } else {
+        survey.ShortName = tempSurvey.ShortName
+    }
+    if survey.LongName != "" {
+        newLongName = survey.LongName
+    } else {
+        survey.LongName = tempSurvey.LongName
+    }
+    if survey.LegalBasis != "" {
+        newLegalBasis = survey.LegalBasis
+    } else {
+        survey.LegalBasis = tempSurvey.LegalBasis
+    }
+    if survey.SurveyMode != "" {
+        newSurveyMode = survey.SurveyMode
+    } else {
+        survey.SurveyMode = tempSurvey.SurveyMode
+    }
+
+    updateQuery := "UPDATE surveyv2.survey SET short_name = $1, long_name = $2, legal_basis = $3, survey_mode = $4 WHERE survey_ref = $5"
+
+    stmt, err := db.Prepare(updateQuery)
+    if err != nil {
+        http.Error(w, "SQL statement not prepared", http.StatusInternalServerError)
+        return
+    }
+
+    defer stmt.Close()
+
+    survey.SurveyRef = params["surveyRef"]
+
+    _, err = stmt.Exec(newShortName, newLongName, newLegalBasis, newSurveyMode, survey.SurveyRef)
+    if err != nil {
+        http.Error(w, "SQL statement error" + err.Error(), http.StatusInternalServerError)
+        return
+    }
+
+    err = tx.Commit()
+    if err != nil {
+            http.Error(w, "Error committing database transaction", http.StatusInternalServerError)
+            return
+        }
+
+    var js []byte
+    js, err = json.Marshal(&survey)
```

Comment:
This won't work with the changes above - we should really be doing another SELECT to ensure we're returning the real state of the data.